### PR TITLE
Transfer Learning (dogs and cats) demo

### PR DESCRIPTION
**Pull request by mwyi**
Uploading the cats dogs demo for easy access

### CODE REVIEW

File: demos/TransferLearning/TransferLearningDemo.py [hunk 1]
```diff
@@ -0,0 +1,481 @@
+import warnings
+warnings.simplefilter('ignore')
+
+import ngraph_bridge
+ngraph_bridge.enable()
+#ngraph_bridge.set_backend('PLAIDML')
```
Comment (dgkutnic) [inline demos/TransferLearning/TransferLearningDemo.py]: Did you mean to leave this commented out?

File: demos/TransferLearning/TransferLearningDemo.py [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+
+            if self.ngraph_backends[i] == 'DISABLED':
+                self.use_ngraph = False
+                ngraph_bridge.disable()
+            else:
+                self.use_ngraph = True
+                ngraph_bridge.enable()
+                ngraph_bridge.set_backend(self.ngraph_backends[i])
+
+    def init_gui(self):
```
Comment (dgkutnic) [inline demos/TransferLearning/TransferLearningDemo.py]: If this is a command-line version of the demo, does the gui stuff still work? Is this function still needed/relevant?

File: demos/TransferLearning/TransferLearningDemo.py (hunk 1 above)
Comment (mwyi) [reply demos/TransferLearning/TransferLearningDemo.py]: Intended to delete that line.

File: demos/TransferLearning/TransferLearningDemo.py (hunk 2 above)
Comment (mwyi) [reply demos/TransferLearning/TransferLearningDemo.py]: This is intended for use in both a notebook as well as CI integration for nGraph. The gui call won't actually do anything if testing is enabled.

File: demos/TransferLearning/TransferLearningDemo.py [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+from tensorflow.keras.applications import ResNet50
+from tensorflow.keras.applications import MobileNetV2
+from tensorflow.keras.models import Sequential
+from tensorflow.python.keras import optimizers
+
+from keras.applications.resnet50 import preprocess_input
+from keras.preprocessing.image import ImageDataGenerator
+
+import ipywidgets as widgets
+from ipywidgets import VBox
```
Comment (earhart) [inline demos/TransferLearning/TransferLearningDemo.py]: Just curious -- why import this one directly, and everything else as `widgets.Thing`?

File: demos/TransferLearning/TransferLearningDemo.py (hunk 1 above)
Comment (earhart) [reply demos/TransferLearning/TransferLearningDemo.py]: There're some other commented bits later on... might need more cleanup?

File: demos/TransferLearning/TransferLearningDemo.py [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+        self.wrong_guesses = []
+
+        display(self.init_gui())
+        self.train_button.disabled = False
+        self.fine_tune_button.disabled = False
+
+        self.init_model()
+
+    def init_images(self):
+        zip_file = tf.keras.utils.get_file(
```
Comment (earhart) [inline demos/TransferLearning/TransferLearningDemo.py]: Just curious -- who owns this bucket?  Is it going to remain stable?

Comment (mwyi) [reply demos/TransferLearning/TransferLearningDemo.py]: The dataset is a subset of Kaggle's cats and dogs dataset; this zip is referenced by the tensorflow tutorial. Will look into maintaining our own dataset though.

File: demos/TransferLearning/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
```
Comment (dgkutnic) [inline demos/TransferLearning/README.md]: I don't think your `README` is formatted the way you expect it to be in GitHub.

See https://github.com/plaidml/plaidml/blob/1b10882100a826b8a64a8d97ed3ec7000a6abf0f/demos/TransferLearning/README.md

**Pull request merged**